packages/odata-common/src/request/odata-request.ts

```diff
@@ -227,9 +227,20 @@ export class ODataRequest<RequestConfigT extends ODataRequestConfig> {
       throw Error('The destination cannot be undefined.');
     }
 
-    return executeHttpRequest(destination, await this.requestConfig(), {
-      fetchCsrfToken: this.config.fetchCsrfToken
-    }).catch(error => {
+    const executeHttpRequestHandler = async () =>
+      executeHttpRequest(destination, await this.requestConfig(), {
```

Comment:
I would pass the middlewares down to the executeHttpRequest in the last argument as indicated in the ADR. Doing so needs to wrap the call only once for all use case: openapi, odata and direct usage.